utils/exceptions.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <QException>
+#include <QDebug>
+
+namespace jbrush
+{
+// Inherits from QException and allows the client to specify a unique string message that
+// is displayed in Qt's debugging console when raised
+class GeneralException : public QException
+{
+protected:
+    QString message;
+public:
+    GeneralException() { message = "Unidentified exception ocurred"; }
+    GeneralException(QString m) { message = m; }
+    void raise() const override
```

Comment:
Why do you use ``qDebug() << message;`` in this function. ``QException`` documentation says to just put ``throw *this;``

Reply:
```suggestion
    void raise() const override {throw *this;}
```